src/test/java/com/serenitydojo/fruitmarket/CatalogTest.java

```diff
@@ -1,30 +1,150 @@
-        //         new CatalogItem(Banana, 1)
-        // );
-        // List<CatalogItem> availableFruits = catalog.getAvailableFruits();
-        // assertThat(availableFruits.get(0).getFruit()).isEqualTo(Apple);
-        // assertThat(availableFruits.get(1).getFruit()).isEqualTo(Banana);
-        // assertThat(availableFruits.get(2).getFruit()).isEqualTo(Pear);
+// TODO: Uncomment this code and make it work - Done
+         List<CatalogItem> availableFruits = catalog.getAvailableFruits();
+         System.out.println("Available fruits in alphabetical order: " + availableFruits);
+         assertThat(availableFruits.get(0).getFruit()).isEqualTo(Apple);
+         assertThat(availableFruits.get(1).getFruit()).isEqualTo(Banana);
+         assertThat(availableFruits.get(2).getFruit()).isEqualTo(Orange);
+         assertThat(availableFruits.get(3).getFruit()).isEqualTo(Pear);
+    }
+
+    @Test(expected = FruitUnavailableException.class )
+    public void shouldReportFruitPrice() throws FruitUnavailableException{
+        Assertions.assertThat(catalog.getUnitPriceOf(Apple)).isEqualTo(4.00);
+        Assertions.assertThat(catalog.getUnitPriceOf(Orange)).isEqualTo(5.50);
+        Assertions.assertThat(catalog.getUnitPriceOf(Banana)).isEqualTo(6.00);
+        Assertions.assertThat(catalog.getUnitPriceOf(Pear)).isEqualTo(4.50);
+        Assertions.assertThat(catalog.getUnitPriceOf(Grapefruit)).isEqualTo(4.50);
+    }
+
+    @Test(expected = FruitUnavailableException.class)
+    public void throwsExceptionIfFruitIsUnavailable() throws FruitUnavailableException {
+        catalog.checkFruitAvailability(Orange);
+
+        //Ash: throws the expected exception because Grapefruit is not in the Catalog
+        //The exception is hidden away because it is marked as "expected" via the annotation above this Test method's signature
+            catalog.checkFruitAvailability(Grapefruit);
+    }
+
+    @Test
+    public void noExceptionIfFruitIsAvailable() throws FruitUnavailableException {
+        catalog.checkFruitAvailability(Apple);
+    }
+
+    @Test
+    //Ash: test that pricing calculation is correct when quantity > 1 kilo
+    public void shouldCalculatePerFruitPriceForMultipleUnits() throws FruitUnavailableException {
+        Double fruitPriceForQuantity = catalog.calculateFruitPriceForQuantity(Apple, 2);
+        Assertions.assertThat(fruitPriceForQuantity).isEqualTo(8);
+    }
+
+    @Test
+    public void canAddItemsToShoppingCart() throws FruitUnavailableException {
+        ShoppingCart cart = catalog.grabShoppingCart();
+
+        cart.addItem(Apple, 1);
+        cart.addItem(Orange, 2);
+        cart.addItem(Banana, 3);
+
+        Map<Fruit, Integer> actualItemsInCart = cart.showItemsInCart();
+
+        Map<Fruit, Integer> expectedItemsInCart = new HashMap<Fruit, Integer>();
+        expectedItemsInCart.put(Apple, 1);
+        expectedItemsInCart.put(Orange, 2);
+        expectedItemsInCart.put(Banana, 3);
```

Comment:
which java version are you using? newer versions allow you to create maps like
```suggestion
        Map<Fruit, Integer> expectedItemsInCart = Map.of(
            Apple, 1,
            Orange, 2,
            Banana, 3
        )
```

Reply:
True! When I og learnt Maps, I was on an older version. Now that I have  an new version, I should totally take advantage of this ability.